src/ApplicationIO.cpp

```diff
@@ -101,8 +104,10 @@ namespace geopm
             }
             m_is_connected = true;
 
-            m_start_energy_pkg = current_energy_pkg();
-            m_start_energy_dram = current_energy_dram();
+            m_energy_pkg_idx = m_platform_io.push_signal("ENERGY_PACKAGE", GEOPM_DOMAIN_BOARD, 0);
+            m_energy_dram_idx = m_platform_io.push_signal("ENERGY_DRAM", GEOPM_DOMAIN_BOARD, 0);
+            m_start_energy_pkg = m_platform_io.sample(m_energy_pkg_idx);
```

Comment:
This is a non-starter.  We cannot sample at this point in the stack.  We will not be able to push any signals other than the ProfileIO signals.  The right solution is to push_signal() and call read_signal() here.  The call at the end should be to sample().